### PR TITLE
Support IPv6 private topology

### PR DESCRIPTION
**Pull request by johngmyers**

### CODE REVIEW

File: pkg/model/awsmodel/network.go [hunk 1]
```diff
@@ -479,6 +497,17 @@ func (b *NetworkModelBuilder) Build(c *fi.ModelBuilderContext) error {
 		}
 		c.AddTask(r)
 
+		if b.IsIPv6Only() {
```
[the remark this answers is not in the document]
Comment (johngmyers) [reply pkg/model/awsmodel/network.go:500]: No. To get to this point, `allPrivateSubnetsUnmanaged` has to be `false`, so if `b.IsIPv6Only()` then `eigw` will be non-`nil`. There is nothing reasonable for a check to do; a panic is the best option.

File: upup/pkg/fi/cloudup/new_cluster.go [hunk 2]
```diff
@@ -1085,6 +1072,19 @@ func setupTopology(opt *NewClusterOptions, cluster *api.Cluster, allZones sets.S
 		return nil, fmt.Errorf("invalid topology %s", opt.Topology)
 	}
 
+	if opt.IPv6 {
+		cluster.Spec.NonMasqueradeCIDR = "::/0"
+		cluster.Spec.ExternalCloudControllerManager = &api.CloudControllerManagerConfig{}
+		if api.CloudProviderID(cluster.Spec.CloudProvider) == api.CloudProviderAWS {
+			klog.Warningf("IPv6 support is EXPERIMENTAL and can be changed or removed at any time in the future!!!")
```
[the remark this answers is not in the document]
Comment (johngmyers) [reply upup/pkg/fi/cloudup/new_cluster.go:1079]: Possibly, but that's a different PR. Once there's a NAT64 solution, I think we can consider removing the feature flag.

**Review by hakman — APPROVED** [review]

**Pull request merged**